curp/src/error.rs

```diff
@@ -46,31 +39,31 @@ pub enum ProposeError {
     /// The command has already been proposed before
     #[error("duplicated, the cmd might have already been proposed")]
     Duplicated,
-    /// Command execution error
-    #[error("command execution error {0}")]
-    ExecutionError(String),
     /// Command syncing error
     #[error("syncing error {0}")]
-    SyncedError(String),
-    /// Rpc Error reported by madsim
-    #[error("rpc error: {0}")]
-    RpcError(String),
+    SyncedError(SyncError),
     /// Encode error
     #[error("encode error: {0}")]
     EncodeError(String),
 }
 
-impl From<tonic::transport::Error> for ProposeError {
+/// The error met during propose phase
+#[derive(Error, Debug, Serialize, Deserialize)]
+#[allow(clippy::module_name_repetitions)] // this-error generate code false-positive
+#[error("rcp error {0}")]
+pub struct RpcError(String);
```

Comment:
What's the difference between `RpcError` and `RpcServiceError`?

Reply:
`RpcServiceError` is not used by any code, I removed it from `ServerError`.